exercises/nucleotide-count/canonical-data.json

```diff
@@ -17,7 +17,30 @@
           }
         },
         {
-          "description": "strand with repeated nucleotide",
+          "comments" : ["Should check for null-references and assume an empty input."],
+          "description": "counts an undefined DNA strand as empty strand.",
+          "property": "nucleotideCounts",
+          "strand": null,
+          "expected": {
```

Comment:
do i have to write ```null``` or ```"null"```?

Reply:
I don't think null input handling is appropriate to test here.
The point of this exercise is string parsing, not having a string at all defeats the whole purpose of the exercise.

Reply:
To answer the question you asked: `null`

Reply:
I see what you mean. 
But as far as I experienced it, test suits should check for edge cases. null checks are especially important (at least in almost all languages). 
In every educational piece of material the advice is not to only check for usual inputs, but for all possible (equivalent kinds of) inputs. null checks, empty checks and others are always on top of the example lists given. Just take any arbitrary course about QA and this will be part of it.

I think this is especially important, because we are in an educational context. The users should learn from the getgo what it means to write (arguably good) tests (first), because it is still a major problem, even upon seasoned programmers. 
From my own business experience I know that new programmers are completely lost when it comes to writing tests. But also experienced programmers fall in the traps of "I'll do it later", "This is a trivial method", "I can't test this", "There are too many dependencies" etc. Just take any talk or tutorial about writing tests or even TDD and you will see the same old counter-arguments or questions again and again from the comments or the audience. This has a reason: *Writing good tests is hard! Very hard to be exact and it is a skill that needs exercise.* And where better to begin than right from the start.

Reply:
I think this concept deserves wider discussion so I've replied as a new issue: https://github.com/exercism/problem-specifications/issues/902